src/Core/MainLoop.js

```diff
@@ -58,6 +58,14 @@ MainLoop.prototype._update = function _update(view) {
 };
 
 MainLoop.prototype._step = function _step(view) {
+    const willRedraw = this.needsRedraw;
+
+    // Reset internal state before calling _update (so future calls to View.notifyChange()
+    // can properly change it)
+    this.needsRedraw = false;
+    this.renderingState = RENDERING_PAUSED;
+    view._changeSources.clear();
```

Comment:
Why did you move these 2 lines? Should they be called *after* calling `this._update` below?